src/uiSelectDirective.js

```diff
@@ -207,10 +207,11 @@ uis.directive('uiSelect',
         }
 
         // See Click everywhere but here event http://stackoverflow.com/questions/12931369
-        $document.on('click', onDocumentClick);
+        // We adjusted this solution to use the capture phase as event-propagation-mode to avoid a stopPropagation from breaking the functionality.
+        $window.document.addEventListener('click', onDocumentClick, true);
 
         scope.$on('$destroy', function() {
-          $document.off('click', onDocumentClick);
+          $window.document.removeEventListener('click', onDocumentClick);
```

Comment:
If a listener is registered twice, one with capture and one without, remove each one separately. Removal of a capturing listener does not affect a non-capturing version of the same listener, and vice versa.

So you need to add the third `true` parameters.